include/dts-functions.sh

```diff
@@ -1375,6 +1344,137 @@ show_menu() {
+          ;;
+      esac
+      if [ "${SEND_LOGS}" == "true" ]; then
+          # DEPLOY_REPORT variable is used in dasharo-hcl-report to determine
+          # which logs should be printed in the terminal, in the future whole
+          # dts scripting should get some LOGLEVEL and maybe dumping working
+          # logs to file
+          export DEPLOY_REPORT="false"
+          wait_for_network_connection && ${CMD_DASHARO_HCL_REPORT} && logs_sent="1"
+      else
+          export DEPLOY_REPORT="false"
+          ${CMD_DASHARO_HCL_REPORT}
+      fi
+      read -p "Press ENTER to continue."
+
+      return 0
+      ;;
+    "${DASHARO_FIRM_OPT}")
+      if ! check_if_dasharo; then
+        if wait_for_network_connection; then
+          echo "Preparing ..."
+          if [ -z "${logs_sent}" ]; then
+            export SEND_LOGS="true"
+            export DEPLOY_REPORT="true"
+            if ! ${CMD_DASHARO_HCL_REPORT}; then
+              echo -e "Unable to connect to cloud.3mdeb.com for submitting the
+                        \rHCL report. Please recheck your internet connection."
+            else
+              logs_sent="1"
+            fi
+          fi
+        fi
+
+        if [ -n "${logs_sent}" ]; then
+          ${CMD_DASHARO_DEPLOY} install
+        fi
+      else
+        # For NovaCustom TGL laptops with Dasharo version lower than 1.3.0,
+        # we shall run the ec_transition script instead. See:
+        # https://docs.dasharo.com/variants/novacustom_nv4x_tgl/releases/#v130-2022-10-18
+        if [ "$SYSTEM_VENDOR" = "Notebook" ]; then
+            case "$SYSTEM_MODEL" in
+              "NS50_70MU"|"NV4XMB,ME,MZ")
+                compare_versions $DASHARO_VERSION 1.3.0
+                if [ $? -eq 1 ]; then
+                # For Dasharo version lesser than 1.3.0
+                  print_warning "Detected NovaCustom hardware with version < 1.3.0"
+                  print_warning "Need to perform EC transition after which the platform will turn off"
+                  print_warning "Then, please power it on and proceed with update again"
+                  print_warning "EC transition procedure will start in 5 seconds"
+                  sleep 5
+                  ${CMD_EC_TRANSITION}
+                  error_check "Could not perform EC transition"
+                fi
+                # Continue with regular update process for Dasharo version
+                #  greater or equal 1.3.0
+                ;;
+            esac
+        fi
+
+        # Use regular update process for everything else
+        ${CMD_DASHARO_DEPLOY} update
+      fi
+      read -p "Press ENTER to continue."
+
+      return 0
+      ;;
+    "${REST_FIRM_OPT}")
+      if check_if_dasharo; then
+        ${CMD_DASHARO_DEPLOY} restore
+      fi
+      read -p "Press ENTER to continue."
+
+      return 0
+      ;;
+    "${DES_KEYS_OPT}")
+      get_des_creds
+
+      # Verify credentials, start loop over if there is smth wrong:
+      check_des_creds || return 0
```

Comment:
In this function, we check for FW variant available behind subscription, if there is none, the check will fail.

But we can have DTS package to be downloaded, even if there is no firmware for the selected platform.

Reply:
Reworked here 8d9404413d261136125bf337aecac1e7b085c1c6.